lite2/rpc/client.go

```diff
@@ -317,8 +352,36 @@ func (c *Client) TxSearch(query string, prove bool, page, perPage int, orderBy s
 	return c.next.TxSearch(query, prove, page, perPage, orderBy)
 }
 
+// Validators fetches and verifies validators.
+//
+// WARNING: only full validator sets are verified (when length of validators is
+// less than +perPage+. +perPage+ default is 30, max is 100).
 func (c *Client) Validators(height *int64, page, perPage int) (*ctypes.ResultValidators, error) {
-	return c.next.Validators(height, page, perPage)
+	res, err := c.next.Validators(height, page, perPage)
+	if err != nil {
+		return nil, err
+	}
+
+	// Validate res.
+	if res.BlockHeight <= 0 {
+		return nil, errNegOrZeroHeight
+	}
+
+	// Update the light client if we're behind.
+	h, err := c.updateLiteClientIfNeededTo(res.BlockHeight)
```

Comment:
do you think this is the right name for the function. Aren't we just getting a verified header to compare the validator set with

Reply:
but it calls VerifyHeaderAtHeight

Reply:
Yeah I'm not too fussed - I just thought that the function name wasn't as helpful